tests/tests/runner.rs

```diff
@@ -163,13 +168,39 @@ async fn file_data_sources() {
 
     assert_eq!(
         query_res,
-        Some(object! { ipfsFile: object!{ id: id , content: "[]" } })
+        Some(object! { ipfsFile: object!{ id: id.clone() , content: "[]" } })
     );
 
-    // Test loading offchain data sources from DB.
+    // assert whether duplicate datasources are created.
     ctx.provider.stop(ctx.deployment.clone()).await.unwrap();
     let stop_block = test_ptr(2);
     ctx.start_and_sync_to(stop_block).await;
+
+    let store = ctx.store.cheap_clone();
+    let writable = store
+        .writable(ctx.logger.clone(), ctx.deployment.id)
+        .await
+        .unwrap();
+    let datasources = writable.load_dynamic_data_sources(vec![]).await.unwrap();
+    assert!(datasources.len() == 1);
+
+    ctx.provider.stop(ctx.deployment.clone()).await.unwrap();
+    let stop_block = test_ptr(3);
+    ctx.start_and_sync_to(stop_block).await;
+
+    let query_res = ctx
+        .query(&format!(r#"{{ ipfsFile1(id: "{id}") {{ id, content }} }}"#,))
+        .await
+        .unwrap();
+
+    assert_eq!(
+        query_res,
+        Some(object! { ipfsFile1: object!{ id: id , content: "[]" } })
+    );
+
+    ctx.provider.stop(ctx.deployment.clone()).await.unwrap();
+    let stop_block = test_ptr(4);
+    ctx.start_and_sync_to(stop_block).await;
```

Comment:
You check the data source count again at the end of the test, and that the data sources are done.